src/DotVVM.Framework/Resources/Scripts/serialization/date.ts

```diff
@@ -61,9 +77,22 @@ export function serializeTime(date: string | Date | null, convertToUtc: boolean
         date2 = date;
     }
 
-    const h = (date2.getTime() - new Date(1, 0, 1).getTime() / 1000 / 3600) | 0;
-    const mi = padNumber(date2.getMinutes(), 2);
-    const s = padNumber(date2.getSeconds(), 2);
-    const ms = padNumber(date2.getMilliseconds(), 3);
-    return `${h}:${mi}:${s}.${ms}0000`;
+    const ticks = Math.abs(date2.getTime() - timeSpanBaseDate.getTime()) | 0;
+    const hours = (ticks / 1000 / 3600) | 0;
+    const minutes = (ticks / 1000 / 60) | 0;
+    const seconds = (ticks / 1000) | 0;
+    const milliseconds = (ticks % 1000) | 0;
```

Comment:
I'm not sure how often will this fail - esp. with regards to dayling saving time. Since it does not allow me to use 0001-01-01 as the base date, I use 2021-01-01. Your logic says that at 3:00 27.3. is h = 3, while at 3:00 28. 3. h = 2:

```


> (new Date("2021-03-28 03:00:00.0000").getTime() - new Date("2021-01-01 00:00:00.0000")) / 1000 / 3600 % 24
2
> (new Date("2021-03-27 03:00:00.0000").getTime() - new Date("2021-01-01 00:00:00.0000")) / 1000 / 3600 % 24
3
```

Reply:
Actually I don't know what should be the result...